### PR TITLE
Add documentation for gcp-broker-provider

### PR DESCRIPTION
**Pull request by adamwalach**
Changes proposed in this pull request:

- documentation for gcp-broker

### CODE REVIEW

File: docs/service-brokers/helm-broker-service-classes/gcp-broker-provider/docs/plans-details.md [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+```
+kubectl create secret generic gcp-broker-data --from-file=sa-key={filename} --from-literal=project-name=kyma-project --namespace {namespace}
+```
+
+Please note that you have to create a secret in every namespace where the GCP Broker Provider class is provisioned.
+
+In the Service Catalog view click **Google Cloud Platform Service Broker Provider**.
+Provisioning of this class adds GCP Service Broker classes to the Service Catalog in a given namespace.
+
+The service account key created by user is used to
```
Comment (klaudiagrz) [inline docs/service-brokers/helm-broker-service-classes/gcp-broker-provider/docs/plans-details.md]:
```suggestion
The service account key is used to 
```

File: docs/service-brokers/helm-broker-service-classes/gcp-broker-provider/docs/plans-details.md [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+kubectl create secret generic gcp-broker-data --from-file=sa-key={filename} --from-literal=project-name=kyma-project --namespace {namespace}
+```
+
+Please note that you have to create a secret in every namespace where the GCP Broker Provider class is provisioned.
+
+In the Service Catalog view click **Google Cloud Platform Service Broker Provider**.
+Provisioning of this class adds GCP Service Broker classes to the Service Catalog in a given namespace.
+
+The service account key created by user is used to 
+generate service account keys used by brokers installed in different namespaces.
```
Comment (klaudiagrz) [inline docs/service-brokers/helm-broker-service-classes/gcp-broker-provider/docs/plans-details.md]:
```suggestion
generate service account keys used by brokers installed in different Namespaces.
```

File: docs/service-brokers/helm-broker-service-classes/gcp-broker-provider/docs/plans-details.md [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+The generated service account key has a **roles/servicebroker.operator** role and is 
+used during provisioning/deprovisioning/binding/unbinding actions.
```
Comment (klaudiagrz) [inline docs/service-brokers/helm-broker-service-classes/gcp-broker-provider/docs/plans-details.md]:
```suggestion
used during provisioning, deprovisioning, binding, and unbinding actions.
```

Comment (klaudiagrz) [reply docs/service-brokers/helm-broker-service-classes/gcp-broker-provider/docs/plans-details.md]: Ok, but we mention that binding and unbinding do not apply to this class, don't we?

File: docs/service-brokers/helm-broker-service-classes/gcp-broker-provider/docs/plans-details.md [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+
+![](assets/gcp-broker-key-management.svg)
+
+Flow of the provisioning process is described by following diagram
+
+![GCP Broker Provisioning](assets/gcp-broker-provisioning.svg)
+
+1. User triggers provision action
+2. During the provisioning process new service account and access key are created in **Google Cloud Platform**
+3. After the provisioning process is over the post install job is triggered.
```
Comment (klaudiagrz) [inline docs/service-brokers/helm-broker-service-classes/gcp-broker-provider/docs/plans-details.md]:
```suggestion
3. After the provisioning process, the post-install job is triggered.
```

File: docs/service-brokers/helm-broker-service-classes/gcp-broker-provider/docs/plans-details.md [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+Binding to this Service Class is disabled.
+
+## Deprovisioning
+
+Flow of the deprovisioning process is described by following diagram
+
+
+![GCP Broker Deprovisioning](assets/gcp-broker-deprovisioning.svg)
+
+1. User triggers deprovision action
```
Comment (klaudiagrz) [inline docs/service-brokers/helm-broker-service-classes/gcp-broker-provider/docs/plans-details.md]:
```suggestion
1. The user triggers the deprovisioning action.
```

File: docs/service-brokers/helm-broker-service-classes/gcp-broker-provider/docs/plans-details.md [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+2. If secret is present access keys are removed from **Google Cloud Platform**
```
Comment (klaudiagrz) [inline docs/service-brokers/helm-broker-service-classes/gcp-broker-provider/docs/plans-details.md]:
```suggestion
2. If the Secret is present, access keys are removed from the Google Cloud Platform.
```

File: docs/service-brokers/helm-broker-service-classes/gcp-broker-provider/docs/plans-details.md [hunk 7]
```diff
@@ -0,0 +1,78 @@
...
+used during provisioning, deprovisioning, binding, and unbinding actions.
+
+![](assets/gcp-broker-key-management.svg)
+
+Provisioning and deprovisioning processes use the [GCP Broker Provider](https://github.com/kyma-project/kyma/tree/master/tools/gcp-broker-provider) tool.
+
+The provisioning process flow looks as follows:
+1. The user triggers the provisioning action.
+2. Provisioning script checks if secret exists in a given namespace and downloads it.
+3. gcloud CLI is configured based on values from secret.
```
Comment (klaudiagrz) [inline docs/service-brokers/helm-broker-service-classes/gcp-broker-provider/docs/plans-details.md]:
```suggestion
3. The script configures gcloud CLI basing on values from the Secret.
```

File: docs/service-brokers/helm-broker-service-classes/gcp-broker-provider/docs/plans-details.md [hunk 8]
```diff
@@ -0,0 +1,78 @@
...
+4. [sc](https://github.com/kyma-incubator/k8s-service-catalog) CLI is used to add GCP Broker to Kyma cluster
```
Comment (klaudiagrz) [inline docs/service-brokers/helm-broker-service-classes/gcp-broker-provider/docs/plans-details.md]:
```suggestion
4. The script uses the [sc](https://github.com/kyma-incubator/k8s-service-catalog) CLI to add the GCP Broker to the Kyma cluster.
```

File: docs/service-brokers/helm-broker-service-classes/gcp-broker-provider/docs/plans-details.md [hunk 9]
```diff
@@ -0,0 +1,78 @@
...
+![](assets/gcp-broker-key-management.svg)
+
+Provisioning and deprovisioning processes use the [GCP Broker Provider](https://github.com/kyma-project/kyma/tree/master/tools/gcp-broker-provider) tool.
+
+The provisioning process flow looks as follows:
+1. The user triggers the provisioning action.
+2. Provisioning script checks if secret exists in a given namespace and downloads it.
+3. gcloud CLI is configured based on values from secret.
+4. [sc](https://github.com/kyma-incubator/k8s-service-catalog) CLI is used to add GCP Broker to Kyma cluster 
+5. `sc` CLI creates service account and access key on the Google Cloud Platform.
```
Comment (klaudiagrz) [inline docs/service-brokers/helm-broker-service-classes/gcp-broker-provider/docs/plans-details.md]:
```suggestion
5. The `sc` CLI creates a service account and access key on the Google Cloud Platform.
```

File: docs/service-brokers/helm-broker-service-classes/gcp-broker-provider/docs/plans-details.md [hunk 10]
```diff
@@ -0,0 +1,78 @@
...
+
+Binding to this Service Class is disabled.
+
+## Deprovisioning
+
+The deprovisioning process flow looks as follows:
+1. The user triggers the deprovisioning action.
+2. Provisioning script checks if secret exists in a given namespace and downloads it. 
+   If secret does not exist the `sc` CLI is executed with the `--skip-gcp-integration` parameter (step 6) 
+3. gcloud CLI is configured based on values from secret.
```
Comment (klaudiagrz) [inline docs/service-brokers/helm-broker-service-classes/gcp-broker-provider/docs/plans-details.md]:
```suggestion
3. The script configures gcloud CLI basing on values from the Secret.
```

File: docs/service-brokers/helm-broker-service-classes/gcp-broker-provider/docs/plans-details.md [hunk 11]
```diff
@@ -0,0 +1,76 @@
...
+3. Click **Create service account**, name your account, and click **Create**.
+4. Set the `Project Owner` role.
+5. Click **Create key** and choose `JSON` as a key type.
+6. Save the `JSON` file.
+7. Create a Secret from the JSON file by running this command:
+    ```
+    kubectl create secret generic gcp-broker-data --from-file=sa-key={filename} --from-literal=project-name={gcp-project} --namespace {namespace}
+    ```
+
+8. Click **Done**
```
[the remark this answers is not in the document]
Comment (adamwalach) [reply docs/service-brokers/helm-broker-service-classes/gcp-broker-provider/docs/plans-details.md]: Without it the account is not really created, we made this mistake ourselves

**Pull request merged**